components/StoryImage/StoryImage.tsx

```diff
@@ -0,0 +1,90 @@
+import { cnb } from 'cnbuilder';
+import { AnimateInView } from '../Animate';
+import { Container } from '../Container';
+import { WidthBox, type WidthType } from '../WidthBox';
+import { type PaddingType } from '@/utilities/datasource';
+import { imageAspectRatios, type ImageAspectRatioType } from '@/utilities/datasource';
+import { getProcessedImage } from '@/utilities/getProcessedImage';
+import { getSbImageSize } from '@/utilities/getSbImageSize';
+import { type AnimationType } from '../Animate';
+import * as styles from './StoryImage.styles';
+
+type StoryImageProps = React.HTMLAttributes<HTMLDivElement> & {
+  imageSrc: string;
+  imageFocus?: string;
+  isLoadingEager?: boolean;
+  alt?: string;
+  caption?: React.ReactNode;
+  aspectRatio?: ImageAspectRatioType;
+  isFullHeight?: boolean;
+  boundingWidth?: 'site' | 'full';
+  width?: WidthType;
+  spacingTop?: PaddingType;
+  spacingBottom?: PaddingType;
+  isCaptionInset?: boolean;
+  animation?: AnimationType;
+  delay?: number;
+};
+
+export const StoryImage = ({
+  imageSrc,
+  imageFocus,
+  isLoadingEager,
+  alt,
+  caption,
+  aspectRatio,
+  isFullHeight,
+  boundingWidth = 'full',
+  width,
+  spacingTop,
+  spacingBottom,
+  isCaptionInset,
+  animation = 'none',
+  delay,
+  className,
+  ...props
+}: StoryImageProps) => {
+  const { width: originalWidth, height: originalHeight } = getSbImageSize(imageSrc);
+  const cropSize = styles.imageCrops[aspectRatio];
+  /**
+   * Crop width and height are used for width and height attributes on the img element.
+   * They don't need to be exact as long as the aspect ratio is correct.
+   */
+  const cropWidth = parseInt(cropSize?.split('x')[0], 10);
+  const cropHeight = aspectRatio === 'free'
+    ? Math.round(originalHeight * 2000 / originalWidth)
+    : parseInt(cropSize?.split('x')[1], 10);
+
+  return (
+    <WidthBox
```

Comment:
What are the differences, and rules, for when to use Container vs Grid vs Flexbox vs Widthbox? I am asking because it looks like you're setting up a convention here and it is probably something that should be documented for the next developer.

Reply:
Good point - I'll document them when I come back.

You've seen the other ones so I'll talk about the `WidthBox` 😄 

So in Storyblok, I have several components set up in a way that a user can pick their bounding width and width (in number of columns) without nesting them inside a Section or Grid, e.g., WYSIWYG, Story Image, media embed. The WidthBox maps to these Storyblok options, eg, a user can add a Story Image, then pick a bounding width of "site" (which is the `cc`), then 10 of 12 columns - these two selected options are then passed into the WidthBox.

Reply:
I'd say the `WidthBox` is more useful for a headlessCMS like what we have in this case since it abstracts out some common layout options. If we're building a site without the CMS part then this won't add much.